prefect_docker/worker.py

```diff
@@ -0,0 +1,632 @@
+"""
+<span class="badge-api beta"/>
+
+Module containing the Docker worker used for executing flow runs as Docker containers.
+
+Note this module is in **beta**. The interfaces within may change without notice.
+
+To start a Docker worker, run the following command:
+
+```bash
+prefect worker start --pool 'my-work-pool' --type docker
+```
+
+Replace `my-work-pool` with the name of the work pool you want the worker
+to poll for flow runs.
+
+For more information about work pools and workers,
+checkout out the [Prefect docs](https://docs.prefect.io/concepts/work-pools/).
+"""
+import enum
+import re
+import sys
+import urllib.parse
+import warnings
+from typing import TYPE_CHECKING, Dict, Generator, List, Optional, Tuple
+
+import anyio.abc
+import packaging.version
+import prefect
+from prefect.docker import (
+    format_outlier_version_name,
+    get_prefect_image_name,
+    parse_image_tag,
+)
+from prefect.experimental.workers.base import (
+    BaseJobConfiguration,
+    BaseWorker,
+    BaseWorkerResult,
+)
+from prefect.settings import PREFECT_API_URL
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+from prefect.utilities.importtools import lazy_import
+from pydantic import Field, validator
+from slugify import slugify
+from typing_extensions import Literal
+
+if TYPE_CHECKING:
+    import docker
+    from docker import DockerClient
+    from docker.models.containers import Container
+    from prefect.client.schemas import FlowRun
+    from prefect.server.schemas.core import Flow
+    from prefect.server.schemas.responses import DeploymentResponse
+else:
+    docker = lazy_import("docker")
```

Comment:
I don't think we need to do this from the `prefect-docker` library :)

Reply:
I kept this because here's an entrypoint that loads this module whenever prefect is loaded so that the worker is dispatchable, so it could cause slow import times for prefect if prefect-docker is installed. Now that you mention it, I suppose I should run some tests to see if lazy importing still provides value. I'll report back with my findings.

Reply:
Ah hm that makes sense 👍

Reply:
I didn't see any difference in import time with lazy importing, so I removed it as suggested!